src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -22,18 +22,15 @@ public void add(T value) {
     public void add(T value, int index) {
         if (checkAddIndex(index)) {
             growIfFull();
-            elements = addResize(value, index);
+            elements = addResize(index);
+            elements[index] = value;
             size++;
         }
     }
 
-    private Object[] addResize(T value, int index) {
-        Object[] update = new Object[elements.length];
-        System.arraycopy(elements, 0, update, 0, index);
-        update[index] = value;
-        System.arraycopy(elements, index, update, index + 1,
-                size - index);
-        return update;
+    private Object[] addResize(int index) {
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        return elements;
```

Comment:
We can make this method void, it's not necessary to return elements, because we already update it.
```suggestion
    private void addResize(int index) {
        System.arraycopy(elements, index, elements, index + 1, size - index);
```

Reply:
agree. Also, move this method after the public and give a more appropriate name or use this line in the method 
directly.